vulcan/lib/orchestration.rb

```diff
@@ -16,11 +16,21 @@ def initialize(workflow, session)
     def run_until_done!(storage)
       load_json_inputs!(storage)
       [].tap do |ran|
+        errors = RunErrors.new
         i = 0
-        until (runnables = next_runnable_build_targets(storage)).empty? || (i += 1) > MAX_RUNNABLE
-          run!(storage: storage, build_target: runnables.first)
-          ran << runnables.first
+        num_cells_successful = 1
+        until (runnables = next_runnable_build_targets(storage)).empty? || (i += 1) > MAX_RUNNABLE || num_cells_successful == 0
+          begin
+            num_cells_successful = 0
```

Comment:
on other suggestion that might help:

```
next_runnable = runnables.find { |r| !errors.include?(r) }
break if next_runnable.nil?
```

Reply:
This would allow the loop to natural keep running as long as there is some non errored runnable in a simple way I think.